pkg/services/folder/folderimpl/folder.go

```diff
@@ -861,7 +861,7 @@ func (s *Service) legacyDelete(ctx context.Context, cmd *folder.DeleteFolderComm
 	for _, folderUID := range folderUIDs {
 		// only hard delete the folder representation in the dashboard store
 		// nolint:staticcheck
-		deleteCmd := dashboards.DeleteDashboardCommand{OrgID: cmd.OrgID, UID: folderUID, ForceDeleteFolderRules: cmd.ForceDeleteRules, SkipSoftDeletedDashboards: true}
+		deleteCmd := dashboards.DeleteDashboardCommand{OrgID: cmd.OrgID, UID: folderUID, ForceDeleteFolderRules: cmd.ForceDeleteRules}
```

Comment:
I'm trying to understand why I added this flag but couldn't find it an explanation. The only reason is to change the behavior from the caller (since `dashboardStore.DeleteDashboard` is called from different places)  but, in this case, never hard delete soft deleted dashboards, but it is clearly wrong.

Reply:
Yeah, this looks like something that was maybe lost in translation during a refactor. Was the original intent that when deleting a folder the dashboards within that folder should be soft deleted rather than hard deleted?

Reply:
Yes, that is the intended behavior. We don't support folder restoring, only dashboard restoring. So when a folder contains dashboards, we want to "move" only the dashboards to the trash. The original idea behind this decision was to reduce the scope of the hackathon project and the fact that a folder is something more straightforward to recreate instead of a dashboard that could need hours of work to recreate.